symfony/framework-bundle/3.3/config/services.yaml

```diff
@@ -5,24 +5,23 @@ parameters:
 services:
     # default configuration for services in *this* file
     _defaults:
-        # automatically injects dependencies in your services
-        autowire: true
-        # automatically registers your services as commands, event subscribers, etc.
-        autoconfigure: true
-        # this means you cannot fetch services directly from the container via $container->get()
-        # if you need to do this, you can override this setting on individual services
-        public: false
+        autowire: true      # Automatically injects dependencies in your services.
+        autoconfigure: true # Automatically registers your services as commands, event subscribers, etc.
+        public: false       # Allows optimizing the container by removing unused services; this also means
+                            # fetching services directly from the container via $container->get() won't work.
+                            # The best practice is to be explicit about your dependencies anyway.
 
     # makes classes in src/ available to be used as services
     # this creates a service per class whose id is the fully-qualified class name
     App\:
         resource: '../src/*'
-        # you can exclude directories or files
-        # but if a service is unused, it's removed anyway
         exclude: '../src/{DataFixtures,Entity,Migrations,Tests}'
 
-    # controllers are imported separately to make sure they
-    # have the tag that allows actions to type-hint services
+    # controllers are imported separately to make sure services can be injected
+    # as action arguments even if you don't extend any base controller class
     App\Controller\:
         resource: '../src/Controller'
         tags: ['controller.service_arguments']
+
+    # add more service definitions when explicit configuration is needed
+    # please note that last definitions always *replace* previous ones
```

Comment:
I think this last note is the most important one, as we've already got several reports from ppl who've got that wrong

Reply:
Sadly I don't think this --> *"please note that last definitions always *replace* previous ones"* will be enough to understand the problem. The worst thing is that is a tricky problem and not easy to explain: if you define a controller services bellow the `App\Controller\` config, it's natural to think that your service will inherit everything from above.

Reply:
that's why we absolutely need this line: to hint the user when something doesn't work as one would maybe expected

Reply:
Should we include a commented-out example config below?

```yml
    # Example: override a service so you can explicitly wire a `$isDebug` constructor arg
    # App\Service\ExampleService
    #     arguments:
    #         $isDebug: '%kernel.debug%'
```

Reply:
not sure: we did not add examples for parameters, and i think it was a conscious